deployment/common/proposalutils/mcms_test_helpers.go

```diff
@@ -172,51 +172,66 @@ func SignMCMSProposal(t *testing.T, env deployment.Environment, proposal *mcmsli
 }
 
 // ExecuteMCMSProposalV2 - Executes an MCMS proposal on a chain. For timelock proposal, use ExecuteMCMSTimelockProposalV2 instead.
-func ExecuteMCMSProposalV2(t *testing.T, env deployment.Environment, proposal *mcmslib.Proposal, sel uint64) {
-	t.Log("Executing proposal on chain", sel)
+func ExecuteMCMSProposalV2(t *testing.T, env deployment.Environment, proposal *mcmslib.Proposal) {
+	t.Log("Executing proposal")
 
-	executorsMap := map[types.ChainSelector]sdk.Executor{}
 	encoders, err := proposal.GetEncoders()
 	require.NoError(t, err)
 
-	family, err := chainsel.GetSelectorFamily(sel)
-	require.NoError(t, err)
+	executorsMap := map[types.ChainSelector]sdk.Executor{}
+	chainSelectors := map[types.ChainSelector]struct{}{}
+	for _, op := range proposal.Operations {
+		family, err := chainsel.GetSelectorFamily(uint64(op.ChainSelector))
+		require.NoError(t, err)
 
-	chainSel := types.ChainSelector(sel)
-
-	switch family {
-	case chainsel.FamilyEVM:
-		encoder := encoders[chainSel].(*evm.Encoder)
-		chain := env.Chains[sel]
-		executorsMap[chainSel] = evm.NewExecutor(encoder, chain.Client, chain.DeployerKey)
-	case chainsel.FamilySolana:
-		encoder := encoders[chainSel].(*solana.Encoder)
-		chain := env.SolChains[sel]
-		executorsMap[chainSel] = solana.NewExecutor(encoder, chain.Client, *chain.DeployerKey)
-	default:
-		require.FailNow(t, "unsupported chain family")
+		chainSelectors[op.ChainSelector] = struct{}{}
+
+		switch family {
+		case chainsel.FamilyEVM:
+			encoder := encoders[op.ChainSelector].(*evm.Encoder)
+			executorsMap[op.ChainSelector] = evm.NewExecutor(
+				encoder,
+				env.Chains[uint64(op.ChainSelector)].Client,
+				env.Chains[uint64(op.ChainSelector)].DeployerKey)
+		case chainsel.FamilySolana:
+			encoder := encoders[op.ChainSelector].(*solana.Encoder)
+			executorsMap[op.ChainSelector] = solana.NewExecutor(
+				encoder,
+				env.SolChains[uint64(op.ChainSelector)].Client,
+				*env.SolChains[uint64(op.ChainSelector)].DeployerKey)
+		default:
+			require.FailNow(t, "unsupported chain family")
+		}
 	}
 
 	executable, err := mcmslib.NewExecutable(proposal, executorsMap)
 	require.NoError(t, err)
 
-	root, err := executable.SetRoot(env.GetContext(), chainSel)
-	require.NoError(t, deployment.MaybeDataErr(err))
+	for chainSelector := range chainSelectors {
```

Comment:
nitpick(no-action): i guess we dont need need the `chainSelectors` map as we have `executorsMap`, we could do the same thing
```suggestion
	for chainSelector := range executorsMap {
```
but can argue one is more explicit/obvious on the intention, so i am good both ways

Reply:
resolved in e88d00caa5345d114d2458b7361c9cec62511564